src/functions/artifacts.ts

```diff
@@ -37,6 +38,27 @@ export function makeRepoPath(
   return `projects/${projectId}/locations/${location}/repositories/${repoName}`;
 }
 
+/** For testing purposes */
+
+export const getRepoCache = new Map<string, artifactregistry.Repository>();
```

Comment:
Two things:
1. Why Map? We always use Record, not Map
2. I specifically hate using global cache objects because they tend to cause subtle test errors (and/or cause prod to use cache when it shouldn't). That's why the backend code puts its cache in `context`. I know we're really pressed for time here; can we at least add a TODO to resolve a discussion about this?